src/Stratis.Bitcoin.Tests/P2P/PeerConnectorTests.cs

```diff
@@ -274,5 +436,20 @@ private IConnectionManager CreateConnectionManager(NodeSettings nodeSettings, Co
 
             return connectionManager;
         }
+
+        private static void AddMissingPeersToConnectionManager(PeerConnectorConnectNode peerConnector)
```

Comment:
@thefazzer we really need to stop using reflection in unit tests. If we are forced to its either bad class design or we aren't exposing the relevant properties.

In this case you will see that the PeerConnector gets initialized with the connection manager's `connectedPeers` collection. So in the test you can add peers to a connection manager instance and call initialize on the peer connector. This will then set the connected peers in the connector class.

Reply:
> we really need to stop using reflection in unit tests

I don't think this is correct – it has been agreed while ago that it is the preferable method of doing things

Reply:
It was agreed in one instance but then blew out of proportion. Like I said, it should really really be last resort but then again I can't see how something can't be refactored or redesigned so that it can be better unit tested.

Reply:
It just feels weird that half of us thinks that we agreed that we should use it and yet have does not. We should discuss this on slack or elsewhere separately